src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -263,7 +264,42 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
-							if (t.init(res.exprs, i, getParser().getHasDelayBefore(), res)) {
+							ParserInstance parser = getParser();
+							boolean error = true;
+							if (parser.getCurrentStructure() != null) {
+								if (t.getUsableStructures().length > 0) {
+									for (Class<? extends Structure> clazz : t.getUsableStructures()) {
+										if (parser.isCurrentStructure(clazz)) {
+											error = false;
+											break;
+										}
+									}
+									if (error) {
+										Skript.error("'" + res.expr + "' cannot be used in " + Utils.a(parser.getCurrentStructure().toString(null, false)) + " structure.");
+									}
+								}
+							} else if (t.getUsableStructures().length > 0) {
```

Comment:
could we grab the usable structures in a variable instead?

Reply:
We can indeed, shall do that in a mo

Reply:
Done it for sections too for consistency